rest/src/main/java/org/jboss/pnc/rest/endpoints/BuildConfigurationEndpointImpl.java

```diff
@@ -110,19 +107,38 @@ public Page<BuildConfiguration> getAll(PageParameters pageParams) {
     private void validate(BuildConfiguration buildConfiguration) {
         if (buildConfiguration != null) {
             Map<String, String> parameters = buildConfiguration.getParameters();
-            String buildCategoryKey = BuildConfigurationParameterKeys.BUILD_CATEGORY.name();
-            if (parameters != null && parameters.containsKey(buildCategoryKey)) {
-                String buildCategoryStr = parameters.get(buildCategoryKey);
-                try {
-                    BuildCategory.valueOf(buildCategoryStr.toUpperCase());
-                } catch (Exception ex) {
-                    List<String> categories = Stream.of(BuildCategory.values())
-                            .map(BuildCategory::name)
-                            .collect(Collectors.toList());
-                    throw new InvalidEntityException(
-                            buildCategoryKey + " value '" + buildCategoryStr + "' is invalid. Valid values are: "
-                                    + String.join(", ", categories) + '.');
-                }
+
+            if (parameters != null) {
+                validateParameters(parameters);
+            }
+        }
+    }
+
+    private void validateParameters(Map<String, String> parameters) {
+        String buildCategoryKey = BuildConfigurationParameterKeys.BUILD_CATEGORY.name();
+        if (parameters.containsKey(buildCategoryKey)) {
+            String buildCategoryStr = parameters.get(buildCategoryKey);
+            try {
+                BuildCategory.valueOf(buildCategoryStr.toUpperCase());
+            } catch (Exception ex) {
+                List<String> categories = Stream.of(BuildCategory.values())
+                        .map(BuildCategory::name)
+                        .collect(Collectors.toList());
+                throw new InvalidEntityException(
+                        buildCategoryKey + " value '" + buildCategoryStr + "' is invalid. Valid values are: "
+                                + String.join(", ", categories) + '.');
+            }
+        }
+
+        final int MAX_ALIGNMENT_PARAMETERS_LENGTH = 8192;
```

Comment:
This should be at least a constant (`private static final` in the class), if we don't want it configurable

Reply:
Changed, btw I can check all the BC params to be of that maximal length, not just alignment param, if that's desirable.

Reply:
I don't think we need it for other params.
This one is limited becaues it's pasted to command line, and command line is limited and users may be incentivized sometimes to make it extra long.
Other params probably not limitied, and users are not likly to make them extra long (meybe except for EXTRA_REPOSITORIES)